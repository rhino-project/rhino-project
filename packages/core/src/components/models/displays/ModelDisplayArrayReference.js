import { useCallback } from 'react';
import DisplayArray from 'rhino/components/forms/displays/DisplayArray';
import { useGlobalComponentForAttribute } from 'rhino/hooks/overrides';

// eslint-disable-next-line no-unused-vars
export const ModelDisplayArrayReferenceBase = ({ model, ...props }) => {
  const accessor = useCallback(
    (value) => value?.map((v) => v.display_name)?.join(', '),
    []
  );

  return <DisplayArray accessor={accessor} {...props} />;
};

const ModelDisplayArrayReference = (props) =>
  useGlobalComponentForAttribute(
    'ModelDisplayArrayReference',
    ModelDisplayArrayReferenceBase,
    props
  );

export default ModelDisplayArrayReference;
