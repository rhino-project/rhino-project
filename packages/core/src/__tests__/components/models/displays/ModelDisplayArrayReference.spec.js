import { sharedDisplayTests } from './sharedDisplayTests';
import ModelDisplayArrayReference from 'rhino/components/models/displays/ModelDisplayArrayReference';

describe('ModelDisplayArrayReference', () => {
  sharedDisplayTests(ModelDisplayArrayReference);
});
