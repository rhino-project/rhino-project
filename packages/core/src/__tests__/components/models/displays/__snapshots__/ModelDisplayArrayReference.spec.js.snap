// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ModelDisplayArrayReference > should render the empty text when value is nullish 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    readonly=""
    type="text"
    value="-"
  />
</DocumentFragment>
`;

exports[`ModelDisplayArrayReference > should render the overridden empty text when value is nullish 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    readonly=""
    type="text"
    value="baz"
  />
</DocumentFragment>
`;

exports[`ModelDisplayArrayReference > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelDisplayArrayReference > should render with global override shorthand for model 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;

exports[`ModelDisplayArrayReference > should render with global override shorthand for model and attribute 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
